### DISCUSSION TITLE
BT HID device, custom firmware, hardware buttons good enough for morse code?

### DISCUSSION THREAD

**Discussion by craigcomstock** [pine64/OpenPineBuds#90] [Q&A] [answered]
I think I saw somewhere that by default the pinebuds presented as a HID device. I was looking for an open source BT HID device that I could customize so that the button can be used with morse code.

So I was wondering if this is possible with the firmware here?

Is the hardware button itself reliable and good enough to handle lots of pressing involved in morse code and responsive enough to detect the timing of say slow morse code (I am just learning morse code).

I read on a reddit thread I think that the buttons were somewhat unresponsive, like reacted properly about 25% of the time so just wanted to check in with folks that have the buds already and maybe have played with firmware.

**Comment by Ralim** [comment, marked as answer]
Hia,
The buds use a capacitive sense button; so you have to do a full finger off- pause-on cycle to detect.

So they are likely too slow for this use case if I'm being honest :/

**Reply by craigcomstock** [reply]
Perfect, that is exactly the detail I needed. Not suitable. Too bad but thanks a ton for this work. Be well.
